index.js

```diff
@@ -96,13 +96,13 @@ async function processPullRequest (pullRequest, context) {
   const response = await apiHelper.getPRFiles(context, number)
   const filesDownloadedPromise = response
     .map(async filename => {
-      const headRevision = apiHelper.getContents(context, filename, ref, pullRequest.head)
+      const headRevision = await apiHelper.getContents(context, filename, ref, pullRequest.head)
```

Comment:
It doesn't look like `getContents()` is an `async` function, does it return a promise?

Reply:
Yes, it returns a promise.
All "GET" calls to GitHub needs await. The reason is that we can have a slow network connection/server.

When we use "POST" call to GitHub initially when we create a check_run we don't need the result immediately and that's why we await later (on line 70).

Reply:
The requirement for `await` makes sense, I just couldn't see any syntax indicating that `getContents()` returns a promise (the function isn't annotated with the `async` keyword and the octokit/rest docs don't state that it returns a Promise either) - as I'm a JS novice I wanted to check.

Reply:
Yep, better ask because there can be a problem there next time.
When I look at the documentation from VS code I see this:
![image](https://user-images.githubusercontent.com/16246778/56890937-879d3e00-6a83-11e9-80bd-f4c853e4e8a3.png)
so, at least from my side, it looks well documented.

Reply:
Nice. I should try and find where VSCode is pulling those docs from.